### PR TITLE
dropdown menu complete

### PR DESCRIPTION
**Pull request by davodey**

### CODE REVIEW

**Comment by davodey** [comment]
Most of the deleted files are outdated react components...please ignore.

@pauljadam please review the dropdown and menu mustache files.

**Comment by davodey** [comment]
The menu.js file is last, below all the removed junk.

**Comment by pauljadam** [comment]
The drop down should probably follow the ARIA Menu Button pattern, http://w3c.github.io/aria-practices/#menubutton

WAI-ARIA Roles, States, and Properties

The element that opens the menu has role button.
The element with role button has aria-haspopup set to true.
When the menu is displayed, the element with role button has aria-expanded set to true. When the menu is hidden, it is recommended that aria-expanded is not present. If aria-expanded is specified when the menu is hidden, it is set to false.
The element that contains the menu items displayed by activating the button has role menu.
Optionally, the element with role button has a value specified for aria-controls that refers to the element with role menu.
Additional roles, states, and properties needed for the menu element are described in 2.16 Menu or Menu bar.


The menu widget should follow the Menu or Menu Bar pattern, http://w3c.github.io/aria-practices/#menu

WAI-ARIA Roles, States, and Properties

A menu is a container of items that represent choices. The element serving as the menu has a role of either menu or menubar.
The items contained in a menu are child elements of the containing menu or menubar and have any of the following roles:
menuitem
menuitemcheckbox
menuitemradio
If activating a menuitem opens a submenu, the menuitem is known as a parent menuitem. A submenu's menu element is contained in or owned by it's parent menuitem.
A parent menuitem has aria-haspopup set to true.
One of the following approaches is used to enable scripts to move focus among items in a menu as described in 4.6 Keyboard Navigation Inside Components:
The menu container has tabindex set to -1 or 0 and aria-activedescendant set to the ID of the focused item.
Each item in the menu has tabindex set to -1, except in a menubar, where the first item has tabindex set to 0.
When a menuitemcheckbox or menuitemradio is checked, aria-checked is set to true.
When a menu item is disabled, aria-disabled is set to true.
Items in a menu may be divided into groups by placing an element with a role of separator between groups. For example, this technique should be used when a menu contains a set of menuitemradio items.
All separators should have aria-orientation consistent with the separator's orientation.
If a menubar is vertically oriented, it has aria-orientation set to vertical. The default value of aria-orientation for a menubar is horizontal.
If a menu is horizontally oriented, it has aria-orientation set to horizontal. The default value of aria-orientation for a menu is vertical.

**Comment by davodey** [comment]
@antelopeb do we need to add the accessibility scripting to the dropdowns?  For our purposes, I don't think we do.  Developers should be either use the component from React or program this themselves.  Thoughts?

File: pattern-lab/source/_patterns/00-ui-kit/06-dropdown/01-dropdown.mustache [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!-- TODO use button component once approved -->
 {{#dropdown}}
 	<div class="dropdown">
-		<button type="button" class="pe-link--btn">Dropdown
+		<button role="button" aria-haspopup="true" aria-expanded="false" type="button" class="pe-link--btn">Dropdown
```
Comment (davodey) [inline pattern-lab/source/_patterns/00-ui-kit/06-dropdown/01-dropdown.mustache]: @pauljadam hows this look?

**Comment by antelopeb** [comment]
@davodey - I think having the accessibility states in there is important. It just keeps us cleaner when we hand this stuff off to other teams :)

**Comment by davodey** [comment]
@antelopeb @jackiejakob @pauljadam take a peek at my latest commit, it includes a few more dropdown button options.  I also gave the drop-down button its own atom in the Buttons area. This will isolate the accessibility options to dropdowns only.

**Comment by jackiejakob** [comment]
Looking good! In the UI kit, it doesn't look like the divider line extends the full width.

**Comment by davodey** [comment]
ugh, I totally missed that.  Thank you!!

**Comment by davodey** [comment]
Thank you Paul!

**Pull request merged**